### PR TITLE
Initial add of elmLS support

### PR DESCRIPTION
**Pull request by megalithic**
I'm not able to fully test this at the moment. These are, roughly, the settings I'm using from coc.nvim.

### CODE REVIEW

File: README_preamble.md [hunk 1]
```diff
@@ -37,6 +37,7 @@ Implemented:
 - [gopls](https://github.com/neovim/nvim-lsp#gopls) (has some errors)
 - [pyls](https://github.com/neovim/nvim-lsp#pyls)
 - [texlab](https://github.com/neovim/nvim-lsp#texlab)
+- [elmls](https://github.com/neovim/nvim-lsp#elmls)
```
Comment (mhinz) [inline README_preamble.md]: Sorted insert, please. :]

File: README.md [hunk 2]
```diff
@@ -37,6 +37,7 @@ Implemented:
 - [gopls](https://github.com/neovim/nvim-lsp#gopls) (has some errors)
 - [pyls](https://github.com/neovim/nvim-lsp#pyls)
 - [texlab](https://github.com/neovim/nvim-lsp#texlab)
+- [elmls](https://github.com/neovim/nvim-lsp#elmls)
```
Comment (mhinz) [inline README.md]: Sorted insert, please!

File: lua/nvim_lsp/elmls.lua [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      elmPath = "elm",
+      elmFormatPath = "elm-format",
+      elmTestPath = "elm-test",
+      elmAnalyseTrigger = "change",
+    };
+    capabilities = default_capabilities;
+    on_init = vim.schedule_wrap(function(client, result)
+      if result.offsetEncoding then
+        client.offset_encoding = result.offsetEncoding
+      end
```
Comment (justinmk) [inline lua/nvim_lsp/elmls.lua]: nit: `vim.schedule_wrap` isn't needed in this nor the clangd one.

Comment (norcalli) [reply lua/nvim_lsp/elmls.lua]: This is true. That was an artifact of when I was printing there before.

**Comment by norcalli** [comment]
Should we add a command to install it? It seems relatively straightforward.

```lua
function skeleton.elmls.install()
  if not vim.fn.executable("elm-language-server") then
    os.execute("npm install -g elm elm-test elm-format @elm-tooling/elm-language-server")
  end
end
```

**Comment by megalithic** [comment]
sounds good. thank you! would have LOVED to have a self-installing deps step though. :)

**Pull request closed without merging**